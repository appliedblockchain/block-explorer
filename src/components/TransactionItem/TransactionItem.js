import React from 'react'
import PropTypes from 'prop-types'
import { formatNumber } from 'utils/formatNumber'
import LabelValuePair from 'components/LabelValuePair'
import JsonViewer from 'components/JsonViewer'
import styles from './style.module.css'

const TransactionItem = props =>
  <div className={styles.transactionItem}>
    <LabelValuePair label="Hash">
      {props.hash}
    </LabelValuePair>

    <LabelValuePair label="From">
      {props.from}
    </LabelValuePair>

    <LabelValuePair label="Gas">
      {formatNumber(props.gas)}
    </LabelValuePair>

    <LabelValuePair label="Input Data" isCode>
      {props.input}
    </LabelValuePair>

    <LabelValuePair label="Decoded Transaction">
      <JsonViewer jsonObject={props.decodedTx}/>
    </LabelValuePair>

    <LabelValuePair label="Decrypted Data">
      <JsonViewer jsonObject={props.decryptedTx}/>
    </LabelValuePair>
  </div>

TransactionItem.propTypes = {
  hash: PropTypes.string,
  from: PropTypes.string,
  gas: PropTypes.number,
  input: PropTypes.string,
  decodedTx: PropTypes.object,
  decryptedTx: PropTypes.object
}

export default TransactionItem
