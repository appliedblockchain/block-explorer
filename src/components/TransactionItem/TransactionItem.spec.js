import React from 'react'
import renderer from 'react-test-renderer'
import TransactionItem from './TransactionItem'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <TransactionItem
        hash={'hashstring'}
        from={'from'}
        gas={123456}
        input={'input'}
        decodedTx={{ foo: 'bar' }}
        decryptedTx={{ foo: 'bar' }}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

