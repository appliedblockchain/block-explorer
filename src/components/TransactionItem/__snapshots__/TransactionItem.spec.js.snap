// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="transactionItem"
>
  <div
    className="labelValuePair"
  >
    <div
      className="labelValuePairItem"
    >
      <div
        className="labelValuePairLabel"
      >
        Hash
      </div>
      <div
        className="labelValuePairValue"
      >
        hashstring
      </div>
    </div>
  </div>
  <div
    className="labelValuePair"
  >
    <div
      className="labelValuePairItem"
    >
      <div
        className="labelValuePairLabel"
      >
        From
      </div>
      <div
        className="labelValuePairValue"
      >
        from
      </div>
    </div>
  </div>
  <div
    className="labelValuePair"
  >
    <div
      className="labelValuePairItem"
    >
      <div
        className="labelValuePairLabel"
      >
        Gas
      </div>
      <div
        className="labelValuePairValue"
      >
        123,456
      </div>
    </div>
  </div>
  <div
    className="labelValuePair"
  >
    <div
      className="labelValuePairItem"
    >
      <div
        className="labelValuePairLabel"
      >
        Input Data
      </div>
      <div
        className="labelValuePairValue codeValue"
      >
        input
      </div>
    </div>
  </div>
  <div
    className="labelValuePair"
  >
    <div
      className="labelValuePairItem"
    >
      <div
        className="labelValuePairLabel"
      >
        Decoded Transaction
      </div>
      <div
        className="labelValuePairValue"
      >
        <div
          className="react-json-view"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "cursor": "default",
              "fontFamily": "monospace",
              "position": "relative",
            }
          }
        >
          <div
            className="pretty-json-container object-container"
          >
            <div
              className="object-content"
            >
              <div
                className="object-key-val"
              >
                <span>
                  <span
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                      }
                    }
                  >
                    <div
                      className="icon-container"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "17px",
                        }
                      }
                    >
                      <span
                        className="expanded-icon"
                      >
                        <svg
                          fill="currentColor"
                          style={
                            Object {
                              "color": "#586e75",
                              "height": "1em",
                              "paddingLeft": "2px",
                              "verticalAlign": "top",
                              "width": "1em",
                            }
                          }
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 5l6 6 6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span />
                    <span
                      style={
                        Object {
                          "color": "#002b36",
                          "cursor": "pointer",
                          "display": "inline-block",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      {
                    </span>
                  </span>
                  <div
                    className="object-meta-data"
                    onClick={[Function]}
                    style={
                      Object {
                        "display": "inline-block",
                        "padding": "0px 0px 0px 10px",
                      }
                    }
                  >
                    <span
                      className="object-size"
                      style={
                        Object {
                          "borderRadius": "3px",
                          "color": "rgba(0, 0, 0, 0.3)",
                          "cursor": "default",
                          "fontStyle": "italic",
                          "margin": "0px 6px 0px 0px",
                        }
                      }
                    >
                      1
                       item
                      
                    </span>
                    <span
                      className="copy-to-clipboard-container"
                    >
                      <span
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "inline",
                          }
                        }
                      >
                        <span
                          className="copy-icon"
                        >
                          <svg
                            fill="currentColor"
                            preserveAspectRatio="xMidYMid meet"
                            style={
                              Object {
                                "color": "#268bd2",
                                "fontSize": "15px",
                                "height": "1em",
                                "marginRight": "3px",
                                "verticalAlign": "top",
                                "width": "1em",
                              }
                            }
                            viewBox="0 0 40 40"
                          >
                            <g>
                              <path
                                d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                              />
                            </g>
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </span>
                <div
                  className="pushed-content object-container"
                >
                  <div
                    className="object-content"
                    style={
                      Object {
                        "marginLeft": "6px",
                      }
                    }
                  >
                    <div
                      className="variable-row"
                      style={
                        Object {
                          ":hover": Object {
                            "borderLeft": "2px solid rgb(235, 235, 235)",
                            "paddingLeft": "19px",
                          },
                          "borderLeft": "1px solid rgb(235, 235, 235)",
                          "paddingBottom": "3px",
                          "paddingLeft": 20,
                          "paddingRight": "5px",
                          "paddingTop": "3px",
                        }
                      }
                    >
                      <span>
                        <span
                          className="object-key"
                          style={
                            Object {
                              ":hover": Object {
                                "opacity": "1",
                              },
                              "color": "#002b36",
                              "display": "inline-block",
                              "fontStyle": "none",
                              "letterSpacing": "0.5px",
                              "opacity": "0.85",
                              "verticalAlign": "top",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "verticalAlign": "top",
                              }
                            }
                          >
                            "
                          </span>
                          <span
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            foo
                          </span>
                          <span
                            style={
                              Object {
                                "verticalAlign": "top",
                              }
                            }
                          >
                            "
                          </span>
                        </span>
                        <span
                          style={
                            Object {
                              "color": "#002b36",
                              "display": "inline-block",
                              "margin": "0px 5px",
                              "verticalAlign": "top",
                            }
                          }
                        >
                          :
                        </span>
                      </span>
                      <div
                        className="variable-value"
                        onClick={null}
                        style={
                          Object {
                            "cursor": "default",
                            "display": "inline-block",
                            "paddingRight": "6px",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "color": "#cb4b16",
                              "display": "inline-block",
                            }
                          }
                        >
                          <span
                            className="string-value"
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "default",
                              }
                            }
                          >
                            "
                            bar
                            "
                          </span>
                        </div>
                      </div>
                      <span
                        className="copy-to-clipboard-container"
                      >
                        <span
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                              "display": "inline",
                            }
                          }
                        >
                          <span
                            className="copy-icon"
                          >
                            <svg
                              fill="currentColor"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "color": "#268bd2",
                                  "fontSize": "15px",
                                  "height": "1em",
                                  "marginRight": "3px",
                                  "verticalAlign": "top",
                                  "width": "1em",
                                }
                              }
                              viewBox="0 0 40 40"
                            >
                              <g>
                                <path
                                  d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                />
                              </g>
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  className="brace-row"
                >
                  <span
                    style={
                      Object {
                        "color": "#002b36",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontWeight": "bold",
                        "paddingLeft": "3px",
                      }
                    }
                  >
                    }
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="labelValuePair"
  >
    <div
      className="labelValuePairItem"
    >
      <div
        className="labelValuePairLabel"
      >
        Decrypted Data
      </div>
      <div
        className="labelValuePairValue"
      >
        <div
          className="react-json-view"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "cursor": "default",
              "fontFamily": "monospace",
              "position": "relative",
            }
          }
        >
          <div
            className="pretty-json-container object-container"
          >
            <div
              className="object-content"
            >
              <div
                className="object-key-val"
              >
                <span>
                  <span
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "inline-block",
                      }
                    }
                  >
                    <div
                      className="icon-container"
                      style={
                        Object {
                          "display": "inline-block",
                          "width": "17px",
                        }
                      }
                    >
                      <span
                        className="expanded-icon"
                      >
                        <svg
                          fill="currentColor"
                          style={
                            Object {
                              "color": "#586e75",
                              "height": "1em",
                              "paddingLeft": "2px",
                              "verticalAlign": "top",
                              "width": "1em",
                            }
                          }
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 5l6 6 6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span />
                    <span
                      style={
                        Object {
                          "color": "#002b36",
                          "cursor": "pointer",
                          "display": "inline-block",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      {
                    </span>
                  </span>
                  <div
                    className="object-meta-data"
                    onClick={[Function]}
                    style={
                      Object {
                        "display": "inline-block",
                        "padding": "0px 0px 0px 10px",
                      }
                    }
                  >
                    <span
                      className="object-size"
                      style={
                        Object {
                          "borderRadius": "3px",
                          "color": "rgba(0, 0, 0, 0.3)",
                          "cursor": "default",
                          "fontStyle": "italic",
                          "margin": "0px 6px 0px 0px",
                        }
                      }
                    >
                      1
                       item
                      
                    </span>
                    <span
                      className="copy-to-clipboard-container"
                    >
                      <span
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "inline",
                          }
                        }
                      >
                        <span
                          className="copy-icon"
                        >
                          <svg
                            fill="currentColor"
                            preserveAspectRatio="xMidYMid meet"
                            style={
                              Object {
                                "color": "#268bd2",
                                "fontSize": "15px",
                                "height": "1em",
                                "marginRight": "3px",
                                "verticalAlign": "top",
                                "width": "1em",
                              }
                            }
                            viewBox="0 0 40 40"
                          >
                            <g>
                              <path
                                d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                              />
                            </g>
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </span>
                <div
                  className="pushed-content object-container"
                >
                  <div
                    className="object-content"
                    style={
                      Object {
                        "marginLeft": "6px",
                      }
                    }
                  >
                    <div
                      className="variable-row"
                      style={
                        Object {
                          ":hover": Object {
                            "borderLeft": "2px solid rgb(235, 235, 235)",
                            "paddingLeft": "19px",
                          },
                          "borderLeft": "1px solid rgb(235, 235, 235)",
                          "paddingBottom": "3px",
                          "paddingLeft": 20,
                          "paddingRight": "5px",
                          "paddingTop": "3px",
                        }
                      }
                    >
                      <span>
                        <span
                          className="object-key"
                          style={
                            Object {
                              ":hover": Object {
                                "opacity": "1",
                              },
                              "color": "#002b36",
                              "display": "inline-block",
                              "fontStyle": "none",
                              "letterSpacing": "0.5px",
                              "opacity": "0.85",
                              "verticalAlign": "top",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "verticalAlign": "top",
                              }
                            }
                          >
                            "
                          </span>
                          <span
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            foo
                          </span>
                          <span
                            style={
                              Object {
                                "verticalAlign": "top",
                              }
                            }
                          >
                            "
                          </span>
                        </span>
                        <span
                          style={
                            Object {
                              "color": "#002b36",
                              "display": "inline-block",
                              "margin": "0px 5px",
                              "verticalAlign": "top",
                            }
                          }
                        >
                          :
                        </span>
                      </span>
                      <div
                        className="variable-value"
                        onClick={null}
                        style={
                          Object {
                            "cursor": "default",
                            "display": "inline-block",
                            "paddingRight": "6px",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "color": "#cb4b16",
                              "display": "inline-block",
                            }
                          }
                        >
                          <span
                            className="string-value"
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "default",
                              }
                            }
                          >
                            "
                            bar
                            "
                          </span>
                        </div>
                      </div>
                      <span
                        className="copy-to-clipboard-container"
                      >
                        <span
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                              "display": "inline",
                            }
                          }
                        >
                          <span
                            className="copy-icon"
                          >
                            <svg
                              fill="currentColor"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "color": "#268bd2",
                                  "fontSize": "15px",
                                  "height": "1em",
                                  "marginRight": "3px",
                                  "verticalAlign": "top",
                                  "width": "1em",
                                }
                              }
                              viewBox="0 0 40 40"
                            >
                              <g>
                                <path
                                  d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                />
                              </g>
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  className="brace-row"
                >
                  <span
                    style={
                      Object {
                        "color": "#002b36",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontWeight": "bold",
                        "paddingLeft": "3px",
                      }
                    }
                  >
                    }
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
